ntbs-service/DataAccess/DataQualityRepository.cs

```diff
@@ -256,7 +256,8 @@ await GetNotificationQueryableForNotifiedTreatmentOutcomeDataQualityAlerts<T>()
             // We can change the EF behaviour by setting UseRelationalNulls to false, however this can only be done
             // globally and in most situations we probably want the default behaviour.
             // See https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.infrastructure.relationaldbcontextoptionsbuilder-2.userelationalnulls?view=efcore-5.0
-            return await _context.NotificationAndDuplicateIds.FromSqlRaw(@"SELECT N1.[NotificationId] AS [NotificationId], N2.[NotificationId] AS [DuplicateId]
+            return await _context.NotificationAndDuplicateIds.FromSqlRaw(@"SELECT N1.[NotificationId] AS [NotificationId], N2.[NotificationId] AS [DuplicateId],
+CASE WHEN P1.[NhsNumber] IS NOT NULL AND P2.[NhsNumber] IS NOT NULL AND P1.[NhsNumber] <> P2.[NhsNumber] THEN CAST(0 AS BIT) ELSE CAST(1 AS BIT) END AS [NhsNumberMatch]
```

Comment:
`NhsNumberMatch` is a confusing name for this field, because there are times when the NHS numbers are different (i.e. one null, one not null) but this field is still true. Maybe something like `NhsNumbersMatchIfBothGiven` or flip the field round and call it something like `TwoDistinctNhsNumbersFound`?

As an aside, I'm a little surprised you chose to implement this logic in the SQL -

Reply:
I've moved the logic to C#